src/core/lib/channel/channel_stack_builder.c

```diff
@@ -124,6 +124,24 @@ bool grpc_channel_stack_builder_move_prev(
   return true;
 }
 
+grpc_channel_stack_builder_iterator *grpc_channel_stack_builder_iterator_find(
+    grpc_channel_stack_builder *builder, const char *filter_name) {
+  GPR_ASSERT(filter_name != NULL);
+  grpc_channel_stack_builder_iterator *it =
+      grpc_channel_stack_builder_create_iterator_at_first(builder);
+  if (grpc_channel_stack_builder_iterator_is_end(it)) return it;
+  do {
+    const char *filter_name_at_it =
+        grpc_channel_stack_builder_iterator_filter_name(it);
+    if (filter_name_at_it != NULL &&
```

Comment:
Oh!  In that case, I suggest replacing the for loop with a while loop:

```
while (grpc_channel_stack_builder_move_next(it)) {
```

Then there's no need to check for the filter name being NULL.  And it's simpler, to boot.